### PR TITLE
Document all manual release steps.

### LINKED ISSUE #139: Known Release Process Issues
I had to make the following changes manually when releasing v1.6:
- [Need to change name of initial release on vX.Y/releases/index.md](https://github.com/projectcalico/calico/pull/138/commits/8a71d7fb56fb34e3c264a8a8e7e9fa44b5a1fc98)
- search for and fix links that appear as: `https://github.com/projectcalico/calico-cni/releases/download/` until #147 is resolved
- [Other remaining steps that need to manually be done](https://github.com/projectcalico/calico/pull/138/commits/9bc04db5462d7ac897141bf76592d4bac6eedf24)

### PR DESCRIPTION
**Pull request by djosborne**
Fixes #139 by documenting all known release issues.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+6. Modify the redirect in `/index.html` to point to your new release.
+
+7. Commit the changes made in steps 2-6.
+
+8. QA the release. An easy way to do this is render the site locally.
+You may also want to push this as a branch on a fork and publish your own github-pages
+site for others to view under a different URL.
+
+9. Opening a pull request will display thousands of additions and be difficult to render in
```
Comment (caseydavenport) [inline RELEASING.md]: I think we still want this to go through the CI that github runs, even if we don't review it as a PR.

The last step should probably be "open a PR, make sure it passes CI, and then merge"

File: RELEASING.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+1. Ensure master docs are passing semaphore tests.
+
+2. Check out latest master locally.
+
+## Creating the release
+
+1. Run `python release-scripts/do_release.py`, which will walk through creation of the necessary
+release directories, and will replace references to nightly artifacts with release ones.
+
+2. Fix calico-cni binary links which are not updated by the release script (see [#147](https://github.com/projectcalico/calico/issues/147)).
```
Comment (caseydavenport) [inline RELEASING.md]: Is there a command we can include that will find all of these links for someone going through this guide?

**Comment by caseydavenport** [comment]
A couple of comments, then LGTM

File: RELEASING.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+3. Modify the new `vX.Y/releases/index.md` with the relevant versioning information
+for this release. Ensure you change the header from `master` on that page as well.
+
+4. Add a section in `_config.yaml` so that `page.version` will be set correctly in the new subdirectory:
+
+   ```
+   -
```
Comment (djosborne) [inline RELEASING.md]: nope, as intended. That's a dash `-`

**Pull request merged**